src/mcp/tools/dataconnect/generate_dataconnect_operation.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { toContent } from "../../util.js";
+import { generateOperation } from "../../../gif/fdcExperience.js";
+
+export const generate_dataconnect_operation = tool(
+  {
+    name: "generate_dataconnect_operation",
+    description: "Generates a Firebase Data Connect Operation based on the deployed app schema.",
+    inputSchema: z.object({
+      prompt: z
+        .string()
+        .describe("A description of the query or mutation operation you would like to generate"),
+      service: z
+        .string()
+        .describe(
+          "The service id or name of the deployed Firebase Data Connect Schema in format: projects/<project-id>/locations/<location>/services/<service-name>",
+        ),
```

Comment:
Should be `.nullish()` and described as optional, saying that the service from firebase.json will be used if none provided. Should also accept just an id without the full name string which should then list services to look up which one it is.

Reply:
Done